skeleton/unitWidthCurveSkeleton.py

```diff
@@ -0,0 +1,205 @@
+            connNeighborsList.append(connNeighbors)
+        # joint point degree > 2 1nd order neighbors are either end points or middle points
+        if set((1, 2)) == set(connNeighborsList) or {1} == set(connNeighborsList) or {2} == set(connNeighborsList):
+            skeletonImLabel[tuple(k)] = 3
+        else:  # crowded points
+            skeletonImLabel[tuple(k)] = 4
+    return skeletonImLabel
+
+
+def getShortestPathSkeleton(skeletonIm):
+    """
+    Return unit width curve/shortest path skeleton
+    Parameters
+    ----------
+    skeletonIm : Numpy array
+        3D binary numpy array of boolean dtype
+
+    Returns
+    -------
+    Numpy array
+        3D binary crowded region removed numpy array of the same shape and dtype
+    """
+    assert (skeletonIm.dtype is bool or np.unique(skeletonIm).tolist() == [0, 1] or
+            np.unique(skeletonIm).tolist() == [0] or
+            np.unique(skeletonIm).tolist() == [1]), "skeletonIm is not boolean it is {}".format(skeletonIm.dtype)
+    if len(skeletonIm.shape) == 2:
+        return skeletonIm
+    else:
+        # initialize
+        start = time.time()
+        se = np.ones([3] * 3, dtype=np.uint8)
+        skeletonImNew = np.zeros_like(skeletonIm, dtype=bool)
+        # compute degrees
+        valencearray = convolve(np.uint8(skeletonIm), TEMPLATE, mode='constant', cval=0)
+        valencearray[skeletonIm == 0] = 0
+        skeletonLabelled = _getAllLabelledArray(skeletonIm, valencearray)
+        # find and locate crowded regions exist
+        # crowded region - 26 connected crowded joint point
+        crowdedRegion = np.zeros_like(skeletonLabelled)
+        crowdedRegion[skeletonLabelled == 4] = 1
+        label, countCrowdedRegions = ndimage.measurements.label(crowdedRegion, structure=se)
+        if np.max(skeletonLabelled) < 4:  # no crowded regions
+            print("no crowded regions")
+            return skeletonIm
+        else:
+            objectify = ndimage.find_objects(label)
+            # detect exits
+            exits = np.logical_or(skeletonLabelled == 1, skeletonLabelled == 2)
+            for crowdRegion in range(countCrowdedRegions):
+                loc = objectify[crowdRegion]
+                # find boundaries of the crowded region
+                zcoords = loc[0]
+                ycoords = loc[1]
+                xcoords = loc[2]
+                regionLowerBoundZ = zcoords.start - 1
+                regionLowerBoundY = ycoords.start - 1
+                regionLowerBoundX = xcoords.start - 1
+                regionUpperBoundZ = zcoords.stop + 1
+                regionUpperBoundY = ycoords.stop + 1
+                regionUpperBoundX = xcoords.stop + 1
+                bounds = [regionLowerBoundZ, regionLowerBoundY, regionLowerBoundX, regionUpperBoundZ, regionUpperBoundY, regionUpperBoundX]
+                # remove negative boundaries
+                bounds = [0 if bound < 0 else bound for bound in bounds]
```

Comment:
Instead of all these lines of code, why not something like this:
bounds = [max(coord.start - 1, 0) for coords in loc] + [max(coord.stop + 1, 0) for coords in loc]

It might be even better to define bounds as a list of three ranges (range(xStart, xStop) etc.) since that's how you always access them below.